spec/blob_spec.rb

```diff
@@ -63,7 +66,62 @@
 
       it { blob.name.should == "テスト.txt" }
       it { blob.data.should include("これはテスト") }
-      it { blob.size.should == 340  }
+      it { blob.size.should == 340 }
+      it { blob.mode.should == 0100755 }
+    end
+  end
+
+  describe 'mode' do
+    context 'file regular' do
+      let(:blob) { Gitlab::Git::Blob.find(
+          repository,
+          'fa1b1e6c004a68b7d8763b86455da9e6b23e36d6',
+          'files/ruby/regex.rb'
+      ) }
+
+      it { blob.name.should == 'regex.rb' }
+      it { blob.path.should == 'files/ruby/regex.rb' }
+      it { blob.size.should == 1200 }
+      it { blob.mode.should == 0100644 }
+    end
+
+    context 'file binary' do
+      let(:blob) { Gitlab::Git::Blob.find(
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.